### PR TITLE
Updated CI.yml to remove running tests on docker

### PR DESCRIPTION
**Pull request by vamshin**
*Issue #, if available:*

*Description of changes:*
Integration tests already run on the ```ubuntu-latest``` machine. We do not need docker to run the integ tests and thus avoiding workflow to depend on ODFE docker image.

Added new action to that could help run sanity on ODFE docker images once we have new release.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/workflows/CI-docker-sanity.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: Build and Test k-NN
+on:
+  push:
+    branches:
```
Comment (jmazanec15) [inline .github/workflows/CI-docker-sanity.yml]: This action will still fail if latest ODFE docker version is unavailable. Is there a way we can check to see if the docker image is available before running the test?

Comment (vamshin) [reply .github/workflows/CI-docker-sanity.yml]: Good catch. For now updated to dummy branch. We will have to fix it to run on demand.
created issue to unblock the release https://github.com/opendistro-for-elasticsearch/k-NN/issues/121

Comment (jmazanec15) [reply .github/workflows/CI-docker-sanity.yml]: Instead of adding it now with dummy branch, I think it might be better off to leave it out of this PR.

Comment (vamshin) [reply .github/workflows/CI-docker-sanity.yml]: makes sense. Removed

**Review by jmazanec15 — APPROVED** [review]
LGTM

**Pull request merged**